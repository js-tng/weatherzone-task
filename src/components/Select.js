import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledSelect = styled.select`
  padding: 6px;
  min-width: 120px;
  font-size: 1.125rem;
  border-radius: 6px;
`

const Select = ({ id, name, placeholder, options, onChange }) => {
  return (
    <StyledSelect id={id} name={name} onChange={onChange} defaultValue={'default'}>
      <option value={'default'} disabled>{placeholder}</option>
      {options.map((option, index) => <option key={`${index}-${option.value}`} value={option.value}>{option.label}</option>)}
    </StyledSelect>
  )
}

Select.defaultProps = {
  placeholder: 'Please select an option'
}

Select.propTypes = {
  /** Sets element id. */
  id: PropTypes.string.isRequired,
  /** Sets element name. */
  name: PropTypes.string.isRequired,
  /** Sets element placeholder. */
  placeholder: PropTypes.string,
  /** Sets element options with label and value. */
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]).isRequired,
  })).isRequired,
  /** Runs function on selection change. */
  onChange: PropTypes.func.isRequired,
}

export default Select