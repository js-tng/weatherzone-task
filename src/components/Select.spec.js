import React from 'react'
import { create } from 'react-test-renderer'
import Select from './Select'

describe('<Select />', () => {
  it('renders Select component', () => {
    const snapshot = create(
      <Select
        id='test-id'
        name='test-name'
        options={[{ label: 'label 1', value: 'value-1' }, { label: 'label 2', value: 'value-2' }]}
        onChange={jest.fn()}
      />).toJSON()

    expect(snapshot).toMatchSnapshot()
  })
})
