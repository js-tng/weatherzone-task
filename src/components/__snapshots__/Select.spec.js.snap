// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders Select component 1`] = `
<select
  className="sc-bdfBwQ llyimA"
  defaultValue="default"
  id="test-id"
  name="test-name"
  onChange={[MockFunction]}
>
  <option
    disabled={true}
    value="default"
  >
    Please select an option
  </option>
  <option
    value="value-1"
  >
    label 1
  </option>
  <option
    value="value-2"
  >
    label 2
  </option>
</select>
`;
